subscriptions/schema.py

```diff
@@ -123,19 +125,60 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         subscription.delete()
 
         logger.info(
-            f"user {user.uuid} unsubscribed child {child.pk} from occurrence "
-            f"{occurrence} free spot notification"
+            f"User {user.uuid} unsubscribed child {child.pk} from occurrence "
+            f"{occurrence} free spot notification."
         )
 
         return UnsubscribeFromFreeSpotNotificationMutation(
             child=child, occurrence=occurrence
         )
 
 
+class UnsubscribeFromAllNotificationsMutation(graphene.relay.ClientIDMutation):
+    """
+    Unsubscribe user from all the notifications.
+
+    NOTE: This mutation deletes the user's FreeSpotNotifications,
+    which are linked to a Child and Occurrence instances.
+    **It should be noted that the current model architecture allows
+    that a child can have multiple guardians, so unsubscribe can delete
+    some notifications from other users as well. However, the UI apps
+    has never allowed more than 1 guardian for a child.**
+    """
+
+    class Input:
+        auth_token = graphene.String(
+            description="Auth token can be used to authorize the action "
+            "without logging in as an user."
+        )
+
+    guardian = graphene.Field(GuardianNode)
+    unsubscribed = graphene.Boolean()
+
+    @classmethod
+    # When the login_required raises a PermissionDenied exception,
+    # use the auth_token from the input variables
+    # to populate the context.user with the token related user
+    # and then try again.
+    @user_from_auth_verification_token(
+        get_token=lambda variables: variables.get("auth_token", None),
+        use_only_when_first_denied=True,
+    )
```

Comment:
As you're using this identical decorator specialization in three places and no other specializations at all then how would you feel about making a specific version for brevity? e.g. in decorators.py:
```python
from functools import partial
...
fallback_to_user_from_auth_verification_token = partial(
    user_from_auth_verification_token(
        get_token=lambda variables: variables.get("auth_token", None),
        use_only_when_first_denied=True,
    ))
```
So then this would be simply `@fallback_to_user_from_auth_verification_token`

Reply:
if for any reason the input would be changed to something else, it would be harder to recognize that also the decorator should be changed. The advantage of this is so small, that I don't think it is a good thing to do. The "specialization" is visible "in purpose".

Reply:
"Explicit is better than implicit" supports keeping them visible in the decorator use. Moving them into a specialization would have been for reducing code duplication, i.e. when at least three places do the exact same thing, then might be a good place to check whether it would be good to move that code somewhere else and use it from there. Having tests that check the decorator specialization and that break if the underlying code is changed would tell if the decorator needs changing. Not a clear cut case i.e. a case where one way is obviously the better way IMHO, so I leave this up to you.